experimental/examples/bubblebench/editable-shared-tree/src/bubblebench.ts

```diff
@@ -0,0 +1,108 @@
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { AppState } from "./appState";
+import {
+    appSchemaData,
+    ClientTreeProxy,
+} from "./schema";
+
+export class Bubblebench extends DataObject {
+    public static get Name() {
+        return "@fluid-example/bubblebench-sharedtree";
+    }
+    private maybeTree?: ISharedTree = undefined;
+    private maybeAppState?: AppState = undefined;
+    static treeFactory = new SharedTreeFactory();
+
+    protected async initializingFirstTime() {
+        this.maybeTree = this.runtime.createChannel(
+            "unique-bubblebench-key-1337",
+            Bubblebench.treeFactory.type
+        ) as ISharedTree;
+
+        this.initializeTree(this.maybeTree);
+
+        this.root.set("unique-bubblebench-key-1337", this.maybeTree.handle);
+    }
+
+    protected async initializingFromExisting() {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.maybeTree = await this.root
+            .get<IFluidHandle<ISharedTree>>("unique-bubblebench-key-1337")!
+            .get();
+    }
+
+    protected async hasInitialized() {
+        if (this.tree === undefined) {
+            throw new Error(
+                "hasInitialized called but tree is still undefined",
+            );
+        }
+        this.maybeAppState = new AppState(
+            this.tree.root as ClientTreeProxy[] & EditableField,
+            /* stageWidth: */ 640,
+            /* stageHeight: */ 480,
+            /* numBubbles: */ 1,
+        );
+
+        const onConnected = () => {
+            // Out of paranoia, we periodically check to see if your client Id has changed and
+            // update the tree if it has.
+            setInterval(() => {
+                const clientId = this.runtime.clientId;
+                if (
+                    clientId !== undefined &&
+                    clientId !== this.appState.localClient.clientId
+                ) {
+                    this.appState.localClient.clientId = clientId;
+                }
+            }, 1000);
+        };
+
+        // Wait for connection to begin checking client Id.
+        if (this.runtime.connected) {
+            onConnected();
+        } else {
+            this.runtime.once("connected", onConnected);
+        }
+    }
+
+    /**
+     * Initialize the schema of the shared tree to that of the Bubblebench AppState
+     * and inserts a node with the initial AppState as the root of the tree.
+     * @param tree - ISharedTree
+     */
+    initializeTree(tree: ISharedTree) {
+        tree.storedSchema.update(appSchemaData);
+    }
+
+    private get tree() {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return this.maybeTree!;
+    }
+
+    public get appState() {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return this.maybeAppState!;
+    }
```

Comment:
I'd prefer the return type to be included on this getter.

Also `!` does not error in the case where the value is not what we expect: I would prefer to actually check for the error here as this is a public API.

Overall having a API that with an inferred return type, which it violates that return type if you call it at the right time (but has no docs to say when that is) isn't great.

Exporting fail from our tree package and doing:

```suggestion
    /**
     * Get the AppState.
     * Cannot be call until after initialization has complected.
     */
    public get appState(): AppState {
        return this.maybeAppState ?? fail("not initialized");
    }
```

would be better.

Reply:
The reason it looks like this was solely to be in line with the rest of the bubble bench implementations.
That being said there is no good reason otherwise so I'm happy to make the change.